### PR TITLE
Remove redundant configuration flags

### PR DESCRIPTION
**Pull request by gacholio**
No longer need to explicitly enable the compressed/full flags -
OMR_GC_POINTER_MODE covers them.

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
FYI @youngar

**Comment by gacholio** [comment]
jenkins test sanity all jdk8

**Comment by youngar** [comment]
Thanks for the heads up! How can you tell which builds are compressed refs?

**Comment by gacholio** [comment]
> Thanks for the heads up! How can you tell which builds are compressed refs?

I think the `xl` variants are compressed, but I'm not sure. I tested every platform internally.

**Comment by gacholio** [comment]
@DanHeidinga The test failure isn't real - failure to create a core in pltest.

**Comment by DanHeidinga** [comment]
`xl` means `large heap`.  The "regular" name is the compressedrefs version

File: runtime/gc_glue_java/configure_includes/configure_linux_390.mk [hunk 1]
```diff
@@ -37,7 +37,6 @@ ifeq (linux_390-64_cmprssptrs_codecov, $(SPEC))
 		--enable-OMRTHREAD_LIB_UNIX \
 		--enable-OMR_ARCH_S390 \
 		--enable-OMR_ENV_DATA64 \
-		--enable-OMR_GC_DOUBLE_MAP_ARRAYLETS \
```
Comment (DanHeidinga) [inline runtime/gc_glue_java/configure_includes/configure_linux_390.mk]: Changing `OMR_GC_DOUBLE_MAP_ARRAYLETS` - intentional?

File: runtime/gc_glue_java/configure_includes/configure_linux_390.mk [hunk 2]
```diff
@@ -37,7 +37,6 @@ ifeq (linux_390-64_cmprssptrs_codecov, $(SPEC))
...
 		--enable-OMR_GC_COMPRESSED_POINTERS \
```
Comment (DanHeidinga) [inline runtime/gc_glue_java/configure_includes/configure_linux_390.mk]:
```suggestion
```

**Comment by youngar** [comment]
@DanHeidinga https://github.com/eclipse/omr/pull/4923 https://github.com/eclipse/openj9/pull/8863

**Comment by DanHeidinga** [comment]
Thanks @youngar.  I would have preferred to see the default be an error rather than "full" if something isn't specified as it makes it way easier to catch missing configuration

File: runtime/gc_glue_java/configure_includes/configure_linux_390.mk (hunk 1 above)
Comment (gacholio) [reply runtime/gc_glue_java/configure_includes/configure_linux_390.mk]: Nope.

Comment (gacholio) [reply runtime/gc_glue_java/configure_includes/configure_linux_390.mk]: Fixed.

**Comment by youngar** [comment]
@DanHeidinga I appreciate that perspective, and I think it points to a problem we have talked a lot about in OMR.  A lot of new users expect to be able to pull and build, without having to specify a heap of configuration options, so we have decided to try to use sane defaults to support them.  At the same time, it almost makes more sense to disable every feature in OMR by default and require configuration when you enable features.

**Comment by youngar** [comment]
lgtm

**Comment by DanHeidinga** [comment]
Given the previous builds passed and the trivial change here, going to merge without rebuilding

**Pull request merged**